BootloaderCommonPkg/Include/Library/CryptoLib.h

```diff
@@ -52,10 +58,31 @@ typedef UINT8 HASH_CTX[IPP_HASH_CTX_SIZE];   //IPP Hash context buffer
 // RSA verify required key parameter format
 //
 typedef struct {
-  UINT32                   Signature;
-  UINT8                    PubKeyData[0];
+  UINT32                   PubExp;
+  UINT8                    Modulus[0];
 } RSA_PUB_KEY;
 
+typedef struct {
+  UINT32                   Identifier;  //signature ('P', 'U', 'B', 'K')
+  UINT16                   KeySize;     //Length of Public Key
+  UINT8                    KeyType;     //RSA or ECC
+  UINT8                    Rsvd;
+  UINT8                    KeyData[0];  //Pubic key data(RSA_PUB_KEY) with KeySize bytes
+} PUB_KEY;
+
+typedef struct {
+  UINT32                   Identifier; //signature Identifier('S', 'I', 'G', 'N')
+  UINT16                   SigSize;    //Length of signature 2K and 3K in bytes
+  UINT8                    SigType;    //PKCSv1.5 or RSA-PSS or ECC
+  UINT8                    HashAlg;    //Hash Alg for signingh SHA256, SHA384
+  UINT8                    Signature[0];  //Signature length defined by SigSize bytes
+} SIGNATURE;
```

Comment:
Rename SIGNATURE  to  SIGNATURE_HDR sine SIGNATURE is a too generic name.

Reply:
Renamed SIGNATURE_HDR